### PR TITLE
Improve GitHub actions

### PR DESCRIPTION
**Pull request by blindpirate**
1. Upgrade conventions plugin to 0.7.5.
2. Add build-receipt.properites.
3. Add a build scan tag to identify GitHub builds.

### CODE REVIEW

File: .github/workflows/contributor-pr.yml [hunk 1]
```diff
@@ -33,9 +33,9 @@ jobs:
         with:
           script: |
             if (context.payload.pull_request && context.payload.pull_request.head.repo.fork) {
-                core.setOutput('sys-prop-args', '-DagreePublicBuildScanTermOfService=yes -DcacheNode=us')
+                core.setOutput('sys-prop-args', '-DagreePublicBuildScanTermOfService=yes -DcacheNode=us -Dscan.tag.GitHubActions')
```
Comment (wolfs) [inline .github/workflows/contributor-pr.yml:36]: We already seem to add the `GH_ACTION` tag. Is there a reason to introduce a new tag? https://ge.gradle.org/s/jt3m4urslkcts

File: .github/workflows/contributor-pr.yml [hunk 2]
```diff
@@ -33,9 +33,9 @@ jobs:
         with:
```
Comment (wolfs) [inline .github/workflows/contributor-pr.yml:33]: While you are at it, should we add the concurrency settings as well? https://github.com/gradle/gradle/issues/19173

**Review by wolfs — COMMENTED** [review]
I left some comments.

File: .github/workflows/contributor-pr.yml (hunk 1 above)
Comment (blindpirate) [reply .github/workflows/contributor-pr.yml]: Yes, removed.

File: .github/workflows/contributor-pr.yml (hunk 2 above)
Comment (blindpirate) [reply .github/workflows/contributor-pr.yml]: PTAL!

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**